themes/default/content/blog/pulumi-security-posture/index.md

```diff
@@ -0,0 +1,96 @@
+
+## Write the Code
+
+Manual process is error prone and difficult to audit, iterate and improve on.  If you’ve created manual resources, consider [importing](https://www.pulumi.com/blog/changes-to-import/) them to Pulumi programs.
+
+## Tags your Stacks
+
+We have previously covered [Tag Policies](https://www.pulumi.com/blog/automatically-enforcing-aws-resource-tagging-policies/) as a best practice for the Cloud Resources themselves, but you can also tag your Pulumi Stacks to help organize and visualize your application resources in logical groupings that will help you assess, respond and automate incident remediation. Here are two examples of using Pulumi’s own Pulumi Provider to tag your Stacks automatically: in [python](https://github.com/pulumi/examples/blob/master/aws-py-stackreference/team/__main__.py#L8-L13) and in [typescript](https://github.com/pulumi/examples/blob/master/aws-ts-stackreference/team/index.ts#L17-L22).
+
+## Deploy from Pipelines
+
+Automate every step of your typical deployment process for anything beyond basic local development from a CI/CD pipeline or triggered by web services. If a human is in the loop, consider their involvement to be limited to the approvals process if needed. Better still is to proactively launch approved resources with Pulumi Multi Language Components and CrossGuard.
+
+Leveraging [Automation API](https://www.pulumi.com/docs/guides/automation-api/) as the programmatic interface for running Pulumi programs without the Pulumi CLI is a strongly typed and safe way to use Pulumi in embedded contexts such as web servers without having to shell out to a CLI. For example, [Elkjop Nordic uses Automation API to provide a Self Service Portal](https://www.pulumi.com/blog/how-elkjop-nordic-enables-developers-to-self-serve-infrastructure/) to build a secure IT Vending Machine full of their application and infrastructure building blocks.
+
+## Log Everything
+
+Track key system events such user and pipeline activity, attempted and restricted activity, and changes to identity and access controls. Use Pulumi Audit logs to simplify this process out of the box.  If applicable, we would suggest [automatically exporting Pulumi Audit Logs](https://www.pulumi.com/docs/intro/pulumi-service/audit-logs/#automated-export) to your systems.
+
+## Use Tokens
+
+Organization Access Tokens, Team Access Tokens and Personal Access Tokens securely connect your automation pipelines and development environments with Pulumi without the risks of association user/password combinations.  Machines talk to Pulumi with Tokens of various types and it is always advisable to use Tokens over Users where possible. More on the types of tokens is below.
+
+## Organizational Access Tokens
```

Comment:
Should this heading be a level deeper under `Use Tokens`, as well as the following headings for Team & Personal access tokens?

Reply:
Made it a deeper level along with Team & Personal Tokens